hibernate-core/src/main/java/org/hibernate/type/CustomType.java

```diff
@@ -100,9 +101,20 @@ public CustomType(UserType<J> userType, String[] registrationKeys, TypeConfigura
 	}
 
 	private JavaType<J> getMappedJavaType(UserType<J> userType) {
-		return userType instanceof UserVersionType<J> userVersionType
-				? new UserTypeVersionJavaTypeWrapper<>( userVersionType )
-				: new UserTypeJavaTypeWrapper<>( userType );
+		if ( userType instanceof JavaType<?> javaType ) {
+			//noinspection unchecked
+			return (JavaType<J>) userType;
+		}
+		else if ( userType instanceof JavaTypedExpressible<?> ) {
+			//noinspection unchecked
+			return ( (JavaTypedExpressible<J>) userType).getExpressibleJavaType();
+		}
+		else if ( userType instanceof UserVersionType<J> userVersionType ) {
+			return new UserTypeVersionJavaTypeWrapper<>( userVersionType );
+		}
```

Comment:
I guess this was a mistake. Should not have deleted this branch. Ooops.

In my defense: no test failed.

Reply:
Oh, forget it, I actually didn't delete that branch haha.